.github/workflows/CI-main.yml

```diff
@@ -12,84 +12,84 @@ on:
 
-            if [[ "$current_year" == "$latest_major_version" && "$current_month" == "$latest_minor_version" ]]; then
-              next_version="${latest_major_version}.${latest_minor_version}.$(($latest_patch_version + 1))"
-            else
-              next_version="${current_year}.${current_month}.0"
-            fi
-          fi
+      - name: Read version info
+        id: version_info
+        run: |
+          # test if dispatch-version was skipped. In that case, creates the same .json file expected by the Upload artifact step
+          if [ ! -f version-info/version_info.json ]; then
+            echo "::warning::version-version.json not found, was the previous step skipped? Creating a new file"
+            json='{
+              "version_number": "${{ inputs.build-number }}",
+              "version_name": "${{ inputs.build-version }}"
+            }'
 
-          if [[ ! -z "${{ inputs.build-number }}" ]]; then
-            echo -e "\nApplying build number override"
-            next_number=${{ inputs.build-number }}
+            # file will be used by the upload step
+            mkdir version-info
+            echo "$json" > version-info/version_info.json
           else
-            echo -e "\nCalculating build number..."
-            next_number=$(($GITHUB_RUN_NUMBER + 1000))
+              echo "::notice::version-version.json found!"
           fi
 
-          echo -e "\n"
-          echo "**Version**: $next_version" | tee -a $GITHUB_STEP_SUMMARY
-          echo "**Build Number**: $next_number" | tee -a $GITHUB_STEP_SUMMARY
-          echo "version=$next_version" >> $GITHUB_OUTPUT
-          echo "build_number=$next_number" >> $GITHUB_OUTPUT
-          echo "xcode_version=$XCODE_VERSION" >> $GITHUB_OUTPUT
+          content=$(cat version-info/version_info.json)
+          echo "version_name=$(echo $content | jq -r .version_name)" >> $GITHUB_OUTPUT
+          echo "version_number=$(echo $content | jq -r .version_number)" >> $GITHUB_OUTPUT
+      - name: Upload version info artifact
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
+        with:
+          name: version-info
+          path: version-info/version_info.json
 
   build:
     name: Build
     needs: resolve-values
-    uses: ./.github/workflows/build.yml
+    uses: bitwarden/ios/.github/workflows/build.yml@main
     strategy:
       matrix:
         variant: [Beta, Production]
     with:
       build-variant: ${{ matrix.variant }}
-      build-version: ${{ needs.resolve-values.outputs.build_version }}
-      build-number: ${{ needs.resolve-values.outputs.build_number }}
+      build-version: ${{ needs.resolve-values.outputs.version_name }}
+      build-number: ${{ needs.resolve-values.outputs.version_number }}
       xcode-version: ${{ needs.resolve-values.outputs.xcode_version }}
```

Comment:
for some reason `${{ env.*** }}` isn't available here, we can access a bunch of other contexts, just not `env`. Seems like an oversight. We'll need to keep routing these through another job for the time being.